src/runtime-rs/crates/resource/src/manager_inner.rs

```diff
@@ -48,15 +52,23 @@ impl ResourceManagerInner {
         toml_config: Arc<TomlConfig>,
     ) -> Result<Self> {
         let cgroups_resource = CgroupsResource::new(sid, &toml_config)?;
+        let hypervisor_name = &toml_config.runtime.hypervisor_name;
+        let block_device_driver = &toml_config
+            .hypervisor
+            .get(hypervisor_name)
+            .context("failed to get hypervisor config")?
+            .blockdev_info
+            .block_device_driver;
         Ok(Self {
             sid: sid.to_string(),
-            toml_config,
+            toml_config: toml_config.clone(),
```

Comment:
Why toml_config needs to be cloned here?

Reply:
fixed

Reply:
fixed